### PR TITLE
Create `lro-location-header` rule to disable `LroLocationHeader` LintDiff rule

### PR DESCRIPTION
**Pull request by tjprescott**
Closes https://github.com/Azure/typespec-azure-pr/issues/3896

**REST API Specs** (9 violations)
https://github.com/Azure/azure-rest-api-specs/pull/28505

### CODE REVIEW

File: packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return {
+      operation: (op: Operation) => {
+        const [httpOperation, _] = getHttpOperation(context.program, op);
+        const responses = httpOperation.responses;
+        for (const response of responses) {
+          if (response.statusCodes !== 202) {
+            continue;
+          }
+          if (response.responses.length !== 1) {
+            throw new Error("Expected exactly one response for 202 status code.");
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts:25]: I agree, but I also don't understand how you could have multiple 202 responses?

[the remark this answers is not in the document]
Comment (tjprescott) [reply packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts:25]: [This](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3Jl3yvIK3Jlc291cmNlLW1hbmFnZXIiOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TRFVbpAKnIG0HEZS5Db3JlzhJSx3dNxnY7CgovKiogQ29udG9zbyDIHiBQcm92aWRlciDmAJ5tZW50IEFQSS4gKi8KQGFybcggTmFtZXNwYWNlCkBzZXJ2aWNlKHsKICB0aXRsZTogIsdYyC5IdWJDbGllbnQiLAp9KQpA5wFgZWQo5wDBcykKbshSIE1pY3Jvc29mdC7SR%2B8AuEFQSSDHTXPkAKNlbnVtIMhUIOQAksQuMjAyMS0xMC0wMS1wcmV2aWV3yDXENCAgQHVzZURlcGVuZGVuY3ko9QEx6AFrcy52MV8wX1DGSF8xKcRAYXJtQ29tbW9u5AGZc8cq10jLKctUNcRIYPIAqWAsCn3mAPNB6AD16wEOIOgCTeQAyG1vZGVsIEVtcGxveWVlIGlzIFRyYWNrZWToAII8yBxQcm9wZXJ0aWVzPukBJOQB0SBvZiBlyEDmARpwYXR0ZXJuKCJeW2EtekEtWjAtOS1dezMsMjR9JCLlAP1rZXkoIsg3xEfGF3BhdGjECHNlZ%2BQCRsojc8Ug5AHvOiBzdHLlArznAOTpAL1w6QCi8gDa6gC%2B6QC9QWfzALxhZ2U%2FOiBpbnQzMjsKxylDaXR50ipjaXR5P%2BoAjccsUHJvZmls01lAZW5jb2RlKCJiYXNlNjR1cmzlANJwxjA%2FOiBieXRlc8lIVGhlIHN0YXR1c8RLdGhlIGxhc3Qg5ADFYXRpb27lA1wgIEB2aXNpYmlsaXR5KCJyZWFkyFXEGOUD41N0YXTEX%2BUCAcwU6QFAxGvMMsV45QDCYekCK8VvQGxyb8Q7dXPmAzPRU%2BUBWy4u6QK%2B0SEs7ADfyFzkAnViZcR56QCGZWTEcCDNYtk8dXBkYXTEP8UzVccO32tkZWxldMhnRMQN2TRjcmXkANxyZXF1ZXN0IGhhxERlbiBhY2NlcMlEQccO6QN76QKXbW92yj7pAplNb3ZlUscV6AKSxHdtb3bkAKBmcm9tIGxvY%2BUB7cZixBPxAnDLM3RvzzF0b%2FoDLOcAlXNwb25z6wQJ5gCWxxbsAJfuAozFPsVkxnzmAhbNbmludGVyZuQFmk%2FoArRzIGV4dGVuZHP2Bl4uyyl7fQrlBjrII8oby1noAM3mA9NvcCBibGFoKCk6IHtA5gCvQ29kZSBfOuQFyywgQGJvZHkgxAXoAKggfSB8IHsg3y3kBA0xNn0KfQo%3D&e=%40azure-tools%2Ftypespec-autorest&options=%7B%7D) might be a way, though it crashes anyways. I'll just update the logic to go through each 202 response.

[the remark this answers is not in the document]
Comment (markcowl) [reply packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts:25]: there should not be, but let's just verify there is at least 1 and take the first

File: packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          }
+          const resp = response.responses[0];
+          if (resp.headers?.["Location"] === undefined) {
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts:28]: Should we do a case-insensitive match here (location or Location?

**Review by markcowl — CHANGES_REQUESTED** [review]

File: packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tjprescott) [reply packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts]: Updated.

**Review by timotheeguerin — APPROVED** [review]

File: docs/libraries/azure-resource-manager/rules/lro-location-header.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+## Asynchronous
+
+Long-running (LRO) operations with 202 responses must have a "Location" response header.
+
+#### ❌ Incorrect
+
+```tsp
+@armResourceOperations
+interface Employees {
+  delete is ArmResourceDeleteAsync<Employee>;
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply docs/libraries/azure-resource-manager/rules/lro-location-header.md]: I updated it to use what the actual problem-generated specs used, which was overriding the `LroHeaders` template parameter with `{}` to get rid of the headers. Generally, using our templates properly should avoid this scenario.

File: packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts (hunk 2 above)
Comment (tjprescott) [reply packages/typespec-azure-resource-manager/src/rules/lro-location-header.ts]: Updated.

**Review by markcowl — APPROVED** [review]

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**